### ISSUE TITLE
Changes after database reconnection are not reflected in the database

### ISSUE THREAD

**Issue by ecobost** [datajoint/datajoint-python#597]
In datajoint version 0.11.2, after the database reconnects I can work on the tables (insert and delete stuff) but the changes won't be reflected in the database neither will changes in the database done by other processes be reflected inside this new tables. For instance
```python3
In [1]: from pipeline import experiment                                                                                                                                                                            
Loading local settings from pipeline_config.json
Connecting ecobost@at-database.ad.bcm.edu:3306

In [2]: experiment.AutoProcessing()                                                                                                                                                                                
Out[2]: 
*animal_id    *session    *scan_idx    priority     autosegment   
+-----------+ +---------+ +----------+ +----------+ +------------+
691           1           1            100          0             
691           2           2            100          0             
691           3           1            100          0             
691           3           2            100          0                        
   ...
 (Total: 972)

In [3]: # WAIT UNTIL SATURDAY/SUNDAY                                                                                                                                                                               

In [4]: experiment.AutoProcessing()
Out[4]: /usr/local/lib/python3.6/dist-packages/datajoint/connection.py:150: UserWarning: Mysql server has gone away. Reconnecting to the server.
  warnings.warn("Mysql server has gone away. Reconnecting to the server.")

*animal_id    *session    *scan_idx    priority     autosegment   
+-----------+ +---------+ +----------+ +----------+ +------------+
691           1           1            100          0             
691           2           2            100          0             
691           3           1            100          0             
691           3           2            100          0                      
   ...
 (Total: 972)

In [5]: (experiment.AutoProcessing() & {'animal_id': 691, 'session': 1, 'scan_idx': 1}).delete_quick()                                                                                                            

In [6]: experiment.AutoProcessing()                                                                                                                                                                               
Out[6]: 
*animal_id    *session    *scan_idx    priority     autosegment   
+-----------+ +---------+ +----------+ +----------+ +------------+
691           2           2            100          0             
691           3           1            100          0             
691           3           2            100          0             
691           4           1            100          0                       
   ...
 (Total: 971)
```
but in a different window:
```python
In [1]: from pipeline import experiment                                                                                                                                                                            
Cannot find configuration settings. Using default configuration. To change that, either
    * modify the local copy of pipeline_config.json that pipeline just saved for you
    * put a file named .pipeline_config.json with the same configuration format in your home
          
Connecting ecobost@at-database.ad.bcm.edu:3306

In [2]: experiment.AutoProcessing()                                                                                                                                                                                
Out[2]: 
*animal_id    *session    *scan_idx    priority     autosegment   
+-----------+ +---------+ +----------+ +----------+ +------------+
691           1           1            100          0             
691           2           2            100          0             
691           3           1            100          0             
691           3           2            100          0                        
   ...
 (972 tuples)
```

**Comment by dimitri-yatsenko** [comment]
Hm.. This may be caused by the database server configured with `autocommit=0`, which means that data manipulation statements accumulate their effect until commit is executed. DataJoint is designed to work with `autocommit=1`.

**Comment by ecobost** [comment]
It does have autocommit off now:
```python
In [54]: dj.conn.connection.query('SHOW VARIABLES LIKE "autocommit";').fetchone()                                                                                                                                  
Out[54]: ('autocommit', 'OFF')
```
It is 'ON' in new connections. Not sure why it would change, datajoint doesn't modify it for new connections or reconnections right?

**Comment by dimitri-yatsenko** [comment]
Found it.  In new connections, datajoint sets autocommit to True. When reconnecting, it does not, allowing the server to set its default.  @guzman-raphael is about to make a PR to fix this.

**Comment by dimitri-yatsenko** [comment]
Fixed in #599
